node/src/types/block.rs

```diff
@@ -1376,21 +1382,23 @@ impl Block {
     pub(crate) fn new_from_header_and_body(
         header: BlockHeader,
         body: BlockBody,
+        merkle_tree_hash_activation: EraId,
     ) -> Result<Self, BlockValidationError> {
-        let hash = header.hash();
+        let hash = header.hash(merkle_tree_hash_activation);
         let block = Block { hash, header, body };
-        block.verify()?;
+        block.verify(merkle_tree_hash_activation)?;
         Ok(block)
     }
 
-    pub(crate) fn header(&self) -> &BlockHeader {
-        &self.header
-    }
-
     pub(crate) fn body(&self) -> &BlockBody {
         &self.body
     }
 
+    /// Returns the reference to the header/
```

Comment:
```suggestion
    /// Returns the reference to the header
```

Reply:
Fixed in https://github.com/casper-network/casper-node/pull/2531/commits/9a4261fae4aa6c6834052ff12514d12d47c9e858